mono/utils/mono-error.c

```diff
@@ -819,6 +826,36 @@ Convert this MonoError to an exception if it's faulty or return NULL.
 The error object is cleant after.
 */
 
+MonoExceptionHandle
+mono_error_convert_to_exception_handle (MonoError *target_error)
+{
+	ERROR_DECL (error);
+
+	HANDLE_FUNCTION_ENTER ()
+
+	MonoExceptionHandle ex = MONO_HANDLE_NEW (MonoException, NULL);
+
+	/* Mempool stored error shouldn't be cleaned up */
+	g_assert (!is_boxed ((MonoErrorInternal*)target_error));
+
+	if (mono_error_ok (target_error))
+		goto exit;
+
+	ex = mono_error_prepare_exception_handle (target_error, error);
+	if (!mono_error_ok (error)) {
+		ERROR_DECL_VALUE (second_chance);
+		/*Try to produce the exception for the second error. FIXME maybe we should log about the original one*/
+		ex = mono_error_prepare_exception_handle (error, &second_chance);
+
+		g_assert (mono_error_ok (&second_chance)); /*We can't reasonable handle double faults, maybe later.*/
```

Comment:
Could use `mono_error_assert_ok` here so it prints the second chance error message.

Reply:
Copy/paste of other code I think.

Reply:
Matches non-handle mono_error_convert_to_exception.

Reply:
Let's remember to fix them both. 😄

Reply:
Open a github issue? :)